lib/pbench/agent/tool_data_sink.py

```diff
@@ -134,21 +136,33 @@ def terminate(self):
         if self.abort_launch:
             return 0
 
-        self.run.terminate()
-        self.run.wait()
+        try:
+            self.run.terminate()
+            self.run.wait()
+        except Exception as exc:
+            self.logger.error(
+                "Failed to terminate expected collector process: '%s'", exc
+            )
+            return 0
         return 1
 
 
 class PromCollector(BaseCollector):
+    """Persistent tool data collector for tools compatible with Prometheus"""
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.volume = self.tool_group_dir / "prometheus"
 
     def launch(self):
 
-        if self.host_tools_dict:
+        if self.host_tools_dict and find_executable("podman"):
             self.abort_launch = False
         else:
+            if not find_executable("podman"):
+                self.logger.error(
+                    "Podman is not installed on this system (required by some registered tools, aborting launch"
+                )
             return 0
 
         config = open("prometheus.yml", "w")
```

Comment:
Consider re-writing this `config` creation sequence using a `with` statement.

Alternatively, you might want create the prometheus.yml file as a Jinja2 template, construct the list of host IP, tool, and port as a python list of tuples, and then invoke Jinja2 to render it to a new file.